sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/ScriptTransformation.scala

```diff
@@ -58,11 +58,12 @@ case class ScriptTransformation(
     child.execute().mapPartitions { iter =>
       val cmd = List("/bin/bash", "-c", script)
       val builder = new ProcessBuilder(cmd)
+      builder.redirectErrorStream(true)
```

Comment:
It's indeed not a good idea to mix the error and output together. Just find a friendly API which can easily redirect the error destination of ProcessBuilder to be the same as the current process which normally should be stderr. 
The idea is the buffer can be consuming immediately to the stderr to prevent the hanging on the input side.